### PR TITLE
Fix SCRAM implementation to adhere spec

### PR DESCRIPTION
**Pull request by tailhook**
By specification `nonce` is not base64 encoded string. Quoting the
spec:

> r: This attribute specifies a sequence of random printable ASCII
>     characters excluding ',' (which forms the nonce used as input to
>     the hash function).  No quoting is applied to this string.

I mean current edgedb doesn't accept anything other than valid
base64-encoded string including padding, which is wrong. (The generation
of nonce using base64 as well as client implementation are fine).

Unfortunately this requires a tiny patch in the edgedb itself (so tests are broken now).

### CODE REVIEW

File: edgedb/scram/__init__.py [hunk 1]
```diff
@@ -187,16 +187,18 @@ def parse_client_first_message(resp: bytes):
     if nonce_attr[0:1] != b'r':
         raise ValueError('malformed SCRAM message')
 
-    _, _, nonce_b64 = nonce_attr.partition(b'=')
-    nonce = base64.b64decode(nonce_b64)
+    _, _, nonce_bin = nonce_attr.partition(b'=')
+    nonce = nonce_bin.decode('ascii')
+    if not nonce.isprintable():
```
Comment (ambv) [inline edgedb/scram/__init__.py]: If you'd like to be thorough, reject commas here, too.

File: edgedb/scram/__init__.py [hunk 2]
```diff
@@ -329,8 +332,10 @@ def parse_server_first_message(msg: bytes):
...
```
Comment (ambv) [inline edgedb/scram/__init__.py]: Ditto.

File: edgedb/scram/__init__.py (hunk 1 above)
Comment (tailhook) [reply edgedb/scram/__init__.py]: The original string is split by commas, so commas are not possible here.

File: edgedb/scram/__init__.py (hunk 2 above)
Comment (tailhook) [reply edgedb/scram/__init__.py]: Answered in the previous comment.

**Comment by Neustradamus** [comment]
@tailhook: Good job!

Linked to:
- https://github.com/scram-sasl/info/issues/1

**Pull request merged**